tests/e2e/tests/simple/simple1_test.go

```diff
@@ -71,33 +74,53 @@ func TestSimpleIngress(t *testing.T) {
 	// with "echo debug server ..." on the /debug uri.
 	url := tc.Kube.IngressOrFail(t) + "/fortio/debug"
 	log.Infof("Fetching '%s'", url)
-	attempts := 7 // should not take more than 1min45s to be live...
-	for i := 1; i <= attempts; i++ {
-		if i > 1 {
-			time.Sleep(15 * time.Second) // wait between retries
-		}
-		resp, err := http.Get(url)
-		if err != nil {
-			log.Warnf("Attempt %d : http.Get error %v", i, err)
-			continue
-		}
 
-		body, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			log.Warnf("Attempt %d : ReadAll error %v", i, err)
-			continue
+	// should not take more than 1min45s to be live...
+	ctx, cancel := context.WithTimeout(context.Background(), 105*time.Second)
```

Comment:
this version is better but it's definitely harder to read and hard to label as a "simple" example to build on

we need to move some of that readiness/correctness to library code so the test is simple to write and read

Reply:
We should definitely consider refactoring this sort of readiness check if its used more than once. It has the nice property of being strictly bounded in time (via `context.WithTimeout`) down through the HTTP path.